drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -620,7 +688,7 @@ static int nrf5_tx(const struct device *dev,
 		nrf_802154_channel_get(), nrf_802154_tx_power_get());
 
 	/* Wait for the callback from the radio driver. */
-	k_sem_take(&nrf5_radio->tx_wait, K_FOREVER);
+	nrf5_callback_sem_take(&nrf5_radio->tx_wait);
```

Comment:
How would this code behave if a transmission with delay larger than `IEEE802154_NRF5_CALLBACK_TIMEOUT_MS` was scheduled? AFAIU the current implementation would throw a fatal error, which is not the intended behavior.

Reply:
You are right, however `IEEE802154_NRF5_CALLBACK_TIMEOUT_MS` is expected to be large. OT requests delayed transmits milliseconds before the transmisios because otherwise whole transmitter is considered busy.

Taking into account `tx_at` value from `nrf5_tx_at` (simillar to ed duration) would be a solution to the problem, however the defensive nature of the timeout would be somehow weakened.
A user can always set `IEEE802154_NRF5_CALLBACK_TIMEOUT_MS` to even higher value.

Do you still think we should take `tx_at` into account (or please propose other in-code-solution)?
Otherwise we could add some warning in the Kconfig help or leave it as it is now.

Reply:
> the defensive nature of the timeout would be somehow weakened.

I think then it would be adequate, not weakened. When an immediate operation such as standalone CCA or transmission are requested, their execution is expected to start immediately. The user specifies with Kconfig that they're willing to wait _x_ milliseconds for such operation to finish. In my opinion the same wait time of _x_ milliseconds should be applied to an operation that starts in an arbitrary moment in time other than "now".

I agree that currently OpenThread only uses small delays when requesting delayed transmission and the default timeout is expected to work correctly in that case. Still I think there's no need to put a limitation on the user if there's no need to do that. In my opinion it would be best to take `tx_at` into account.

Reply:
I think I fixed this, please have a look.

Reply:
Thank you.